libexec/bats-core/bats

```diff
@@ -221,7 +221,9 @@ while [[ "$#" -ne 0 ]]; do
   --gather-test-outputs-in)
     shift
     output_dir="$1"
-    if ! mkdir "$output_dir"; then
+    if [ -d "$output_dir" ] && [ -n "$(ls -A "$output_dir")" ]; then
```

Comment:
I know you discourage invoking substells, but this seemed reasonable given that it would be invoked only once, only at startup, and only if this particular option is used.

Reply:
I think direct usage of find might even allow for a less strict test.